JeMPI_Apps/JeMPI_UI/src/pages/settings/uniqueToGR/UniqueToGR.tsx

```diff
@@ -198,23 +166,40 @@ const UniqueToGR = () => {
         }
       }}
     >
-      <DataGrid
-        rows={rows}
-        columns={columns}
-        editMode="row"
-        rowModesModel={rowModesModel}
-        onRowModesModelChange={handleRowModesModelChange}
-        onRowEditStop={handleRowEditStop}
-        processRowUpdate={processRowUpdate}
-        slots={{
-          toolbar: EditToolbar
-        }}
-        slotProps={{
-          toolbar: { setRows, setRowModesModel }
-        }}
-      />
+      {uniqueToGoldenRecordData  && (
+        <DataGrid
+          rows={rows}
+          columns={columns}
+          editMode="row"
+          rowModesModel={rowModesModel}
+          onRowModesModelChange={handleRowModesModelChange}
+          onRowEditStop={handleRowEditStop}
+          processRowUpdate={processRowUpdate}
+          slots={{
+            toolbar: EditToolbar
+          }}
+          slotProps={{
+            toolbar: { setRows, setRowModesModel }
+          }}
+        />
+      )}
     </Box>
   )
 }
 
+function EditToolbar(props: EditToolbarProps) {
```

Comment:
@NyashaMuusha I notificed that this method is replicated in each tab. Can you refactor and extract this to a util that each tab can reference

Reply:
@NyashaMuusha I think this just needs to be replaced with the shared component